File: app/src/main/java/foodie/app/rubikkube/foodie/model/FoodPivot.java
package foodie.app.rubikkube.foodie.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class FoodPivot implements Serializable {
    @SerializedName("profile_id")
    @Expose
    private String profileId;
    @SerializedName("food_id")
    @Expose
    private String foodID;

    public String getProfileId() { return profileId; }

    public void setProfileId(String profileId) { this.profileId = profileId; }

    public String getFoodID() { return foodID; }

    public void setFoodID(String foodID) { this.foodID = foodID; }
}

File: app/src/main/java/foodie/app/rubikkube/foodie/models/FoodPivot.kt
package foodie.app.rubikkube.foodie.models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class FoodPivot(
    @Expose
    @SerializedName("food_id")
    var foodId: String,

    @Expose
    @SerializedName("profile_id")
    var profileId: String
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString()!!,
        parcel.readString()!!
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(foodId)
        parcel.writeString(profileId)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FoodPivot> {
        override fun createFromParcel(parcel: Parcel): FoodPivot {
            return FoodPivot(parcel)
        }

        override fun newArray(size: Int): Array<FoodPivot?> {
            return arrayOfNulls(size)
        }
    }
}